fitsio-sys/build.rs

```diff
@@ -33,3 +34,91 @@ PKG_CONFIG_PATH=<blah> cargo build
         Err(e) => panic!("Unhandled error: {:?}", e),
     };
 }
+
+#[cfg(feature = "fitsio-src")]
+fn bind_cfitsio() {
+    use std::env::var;
+    use std::path::PathBuf;
+
+    let cfitsio_project_dir = PathBuf::from("ext/cfitsio");
+    if !cfitsio_project_dir.exists() {
+        panic!(
+            "Expected to find cfitsio source directory {}",
+            cfitsio_project_dir.display()
+        );
+    }
+    // Make sure the source directory isn't empty.
+    match std::fs::read_dir(&cfitsio_project_dir) {
+        Ok(mut d) => {
+            if let None = d.next() {
+                panic!("cfitsio source directory ext/cfitsio is empty!");
+            }
+        }
+        _ => panic!("Could not read from cfitsio source directory ext/cfitsio !"),
+    }
+
+    // Translate rustc optimisation levels to things a C compiler can
+    // understand. I don't know if all C compilers agree here, but it should
+    // at least work for gcc.
+    let opt_level: String = match var("OPT_LEVEL").as_ref().map(|o| o.as_str()) {
+        Err(_) => panic!("Something wrong with OPT_LEVEL"),
+        // gcc doesn't handle 'z'. Just set it to 's', which also optimises
+        // for size.
+        Ok("z") => "s",
+        Ok(o) => o,
+    }
+    .to_string();
+
+    // Run the contigure script. I'd use the autotools crate here, but it
+    // always outputs two of --{enable,disable}-{shared,static}, none of
+    // which is supported by the cfitsio configure script! So, just run the
+    // script manually.
+    let dst = PathBuf::from(var("OUT_DIR").unwrap());
+
+    std::process::Command::new("make")
+        .arg("clean")
+        .current_dir(&cfitsio_project_dir)
+        .spawn()
+        .expect("Couldn't run cfitsio make clean")
+        .wait()
+        .expect("Failed to wait on child");
+
+    std::process::Command::new("sh")
+        .args(&[
+            "./configure",
+            &format!("--prefix={}", dst.display()),
+            // cfitsio should always be built with reentrant support.
+            "--enable-reentrant",
+            // The user's reference guide states that using SSSE3 and SSE2
+            // can make reading or writing FITS images 20-30% faster(!).
+            // Enabling SSSE3 and SSE2 could cause portability problems, but
+            // it's unlikely that anyone is using such a CPU...
+            // https://stackoverflow.com/questions/52858556/most-recent-processor-without-support-of-ssse3-instructions
+            "--enable-ssse3",
```

Comment:
Could it be worth adding `-march=native` instead of these options? That might prevent cross compiling but I don't think that's an immediate use case.

Reply:
I was going to speak out against putting `-march=native` here, but now I've changed my mind.  See, I was intending this feature that this whole PR revolves around to provide various Rust-compiled libraries and executables via something like GitHub actions, but, it makes more sense to have GitHub actions to have the portable builds, and any downstream users get full performance out of cfitsio by using this feature.  So yes, good catch!  I'll push these last two changes to `build.rs`.